app/javascript/components/CustomTable.vue

```diff
@@ -1,23 +1,36 @@
 <template>
   <table
-    class="table-auto  sm:bg-white rounded-lg overflow-hidden sm:shadow-lg my-5"
+    class="w-full flex flex-row flex-no-wrap sm:bg-white rounded-lg overflow-hidden sm:shadow-lg "
   >
     <thead class="text-white">
       <slot name="columns">
         <tr
-          class="bg-indigo-500  flex flex-col flex-no wrap sm:table-row rounded-l-lg  mb-2"
+          class="bg-indigo-500 flex flex-col flex-no wrap sm:table-row rounded-l-lg sm:rounded-none mb-2 sm:mb-0"
+          v-for="(column, index) in data"
+          v-bind:key="index"
         >
-          <th  class="p-3 text-left" v-for="(column, index) in columns" v-bind:key="index">
+          <th
+            class="p-3 text-left"
+            v-for="(column, index) in columns"
```

Comment:
me tinca mas ponerle nombres diferentes a los `index`, algo como `dataIndex` y `columnIndex`. Lo digo porque un `v-for` esta dentro de otro y se pierde lógica.